common/events_io.hpp

```diff
@@ -124,6 +124,21 @@ class AsyncWriterFromWriter : virtual public mio::AsyncWriter {
 	EventLoop &loop_;
 };
 
+using AsyncReaderFromEventLoopFunc = function<mio::AsyncReaderPtr(EventLoop &loop)>;
+
+class ReaderFromAsyncReader : virtual public mio::Reader {
+public:
+	// Since the async reader usually needs the event loop object, we need to use this factory
+	// function to create the async reader inside of the constructor.
+	ReaderFromAsyncReader(AsyncReaderFromEventLoopFunc func);
```

Comment:
Shouldn't a factory function be `static`?

Reply:
It is the `AsyncReaderFromEventLoopFunc` which is a factory function, not the `ReaderFromAsyncReader`. I clarified this in the comment.